src/cromshell/status/command.py

```diff
@@ -0,0 +1,124 @@
+import fileinput
+import json
+import logging
+import re
+import sys
+
+import click
+import jq
+import requests
+
+from cromshell.utilities import CromshellConfig, HTTPUtils, IOUtils
+
+LOGGER = logging.getLogger(__name__)
+
+
+@click.command(name="status")
+@click.argument("workflow_id")
+@click.pass_obj
+def main(cromshell_config, workflow_id):
+    """Check the status of a Cromwell job UUID"""
+    if cromshell_config.is_verbose:
+        LOGGER.info("status")
+    else:
+        click.echo("status")
+
+    ret_val = 0
+
+    # Set cromwell server using submission file. Running the function below with passing only the workflow id
+    # overrides the default cromwell url set in the cromshell config file, command line argument, and
+    # environment. This takes place only if the workflow id is found in the submission file.
+    CromshellConfig.CromshellConfig.override_cromwell_config_server(
+        workflow_id=workflow_id
+    )
+
+    # Check if Cromwell Server Backend works
+    HTTPUtils.assert_can_communicate_with_server(cromshell_config.cromwell_server)
+
+    # Request workflow status
+    request_out = requests.get(
+        "{}/api/workflows/v1/{}/status".format(
+            cromshell_config.cromwell_server, workflow_id
+        )
+    )
+
+    requested_status_json = request_out.content
+    workflow_status_description = json.loads(request_out.content)
+
+    # Hold our status string here
+    workflow_status = workflow_status_description["status"]
+
+    # Set return value based on workflow status
+    if workflow_status in ("Failed", "Aborted", "fail"):
+        ret_val = 1
+        IOUtils.turtle_dead(cromshell_config.is_verbose)
+    elif workflow_status == "Succeeded":  # change to Running for the final version.
+        # Status claims this workflow is running fine, but we need to check to see
+        # if there are any failed sub-processes.
+        # To do this, we use the `execution-status-count` logic with some filtering:
+
+        # Get execution status count and filter the metadata down:
+        request_meta_out = requests.get(
+            "{}/api/workflows/v1/{}/metadata?{}".format(
+                cromshell_config.cromwell_server,
+                workflow_id,
+                cromshell_config.slim_metadata_parameters,
+            )
+        )
+
+        execution_status_count = (
+            jq.compile(
```

Comment:
please please get rid of this complex nightmare and replace it with some clearer python processing.